MDANSE/Src/MDANSE/Framework/Jobs/DynamicIncoherentStructureFactor.py

```diff
@@ -271,50 +272,29 @@ def finalize(self):
         self.configuration["q_vectors"]["generator"].write_vectors_to_file(
             self._outputData
         )
-
-        nAtomsPerElement = self.configuration["atom_selection"].get_natoms()
-        weights = self.configuration["weights"].get_weights()
-        weight_dict = get_weights(weights, nAtomsPerElement, 1)
-        assign_weights(self._outputData, weight_dict, "f(q,t)_%s")
-        assign_weights(self._outputData, weight_dict, "s(q,f)_%s")
-        if self.add_ideal_results:
-            assign_weights(self._outputData, weight_dict, "s(q,f)_ideal_%s")
-        for element, number in list(nAtomsPerElement.items()):
-            extra_scaling = 1.0 / number
-            self._outputData[f"f(q,t)_{element}"] *= extra_scaling
-            self._outputData[f"s(q,f)_{element}"][:] = get_spectrum(
-                self._outputData[f"f(q,t)_{element}"],
-                self.configuration["instrument_resolution"]["time_window"],
-                self.configuration["instrument_resolution"]["time_step"],
-                axis=1,
-            )
-            if self.add_ideal_results:
-                self._outputData[f"s(q,f)_ideal_{element}"][:] = get_spectrum(
-                    self._outputData[f"f(q,t)_{element}"],
-                    None,
+        self._groupers["fqt"].finalise_centre_of_mass()
+
+        for inkey in self._outputData.keys():
+            if "f(q,t)" in inkey:
```

Comment:
```suggestion
        for inkey in self._outputData:
            if "f(q,t)" in inkey:
```
Could also write as:
```suggestion 
        for inkey in (key for key in self._outputData if "f(q,t)" in key):
```
And remove a layer of nesting